sphinx_immaterial/apidoc/cpp/api_parser.py

```diff
@@ -0,0 +1,2217 @@
+    for token in cursor.get_tokens():
+        location = token.location
+        break
+    else:
+        location = cursor.location
+    initial_location = location
+    f = location.file
+    line = location.line
+    presumed_file, presumed_line, presumed_column = get_presumed_location(location)
+    end_location = SourceLocation.from_position(translation_unit, f, line, 1)
+    comment_lines = []
+    COMMENT = TokenKind.COMMENT
+    while True:
+        prev_location = SourceLocation.from_position(translation_unit, f, line - 1, 1)
+        # `cast` below is workaround for: https://github.com/tgockel/types-clang/pull/2
+        tokens = translation_unit.get_tokens(
+            extent=SourceRange.from_locations(
+                cast(int, prev_location), cast(int, end_location)
+            )
+        )
+        stop = False
+        got_line = False
+        for token in tokens:
+            if token.location == initial_location:
+                stop = True
+                break
+            if token.kind != COMMENT:
+                stop = True
+                break
+            spelling = token.spelling
+            if not spelling.startswith("///"):
+                stop = True
+                break
+            (
+                new_presumed_file,
+                new_presumed_line,
+                new_presumed_column,
+            ) = get_presumed_location(token.location)
+            if (
+                new_presumed_file != presumed_file
+                or new_presumed_line != presumed_line - 1
+            ):
+                stop = True
+                break
+            presumed_line = new_presumed_line
+            if spelling.startswith("/// "):
+                comment_lines.append(spelling[4:])
+            else:
+                comment_lines.append(spelling[3:])
+            got_line = True
+            break
+        if stop or not got_line:
+            break
+        line = line - 1
+        end_location = prev_location
+    if not comment_lines:
+        return None
+    comment_lines.reverse()
+    return {
+        "text": "\n".join(comment_lines),
+        "location": _get_location_json(config, end_location),
+    }
+
+
+class Extractor:
+    def __init__(self, config: Config):
+        self.config = config
+
+        input_path = config.input_path
+
+        input_content = config.input_content
+        if input_content is None:
+            input_content = pathlib.Path(input_path).read_bytes()
+
+        input_content = re.sub(
+            b"#pragma clang module", b"//#pragma clang module", input_content
+        )
+
+        self.input_source = input_content
+
+        self.index = clang.cindex.Index.create()
+        start_time = time.time()
+        self.tu = self.index.parse(
+            input_path,
+            unsaved_files=[(input_path, input_content)],
+            args=tuple(config.compiler_flags) + ("-ferror-limit=0",),
+            options=(  # TranslationUnit.PARSE_SKIP_FUNCTION_BODIES +
```

Comment:
Should add a comment.  If function bodies are skipped, the extent does not include the body.  That interferes with detecting if there are multiple entities defined with no space between (so that they will all be documented together).

Reply:
Oh ok. Yeah, a comment would've satisfied that inquiry.